src/main/java/roomescape/dao/RoomDAO.java

```diff
@@ -0,0 +1,58 @@
+package roomescape.dao;
+
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+import roomescape.domain.Reservation;
+import roomescape.domain.Time;
+
+import java.util.List;
+
+@Repository
+public class RoomDAO {
+    private final JdbcTemplate jdbcTemplate;
+
+    public RoomDAO(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    public List<Reservation> findAll() {
+        return jdbcTemplate.query("SELECT r.id as reservation_id, r.name, r.date, t.id as time_id, t.time as time_value " +
+                        "FROM reservation as r inner join time as t on r.time_id = t.id",
+                (resultSet, rowNum) -> new Reservation(
+                        resultSet.getInt("reservation_id"),
+                        resultSet.getString("name"),
+                        resultSet.getString("date"),
+                        new Time(resultSet.getInt("time_id"), resultSet.getString("time_value"))
+                ));
+    }
+
+    public void insert(Reservation reservation) {
+        jdbcTemplate.update("INSERT INTO reservation (name, date, time_id) VALUES (?, ?, ?)",
+                reservation.getName(), reservation.getDate(), reservation.getTime().getId());
+    }
+
+    public void delete(int id) {
+        jdbcTemplate.update("DELETE FROM reservation WHERE id = ?", id);
+    }
```

Comment:
```suggestion
    public void deleteById(int id) {
        jdbcTemplate.update("DELETE FROM reservation WHERE id = ?", id);
    }
```
- 메서드명을 좀 더 명확히 작성하면 좋을 것 같아요!